mods/sethome/init.lua

```diff
@@ -24,42 +25,58 @@ loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
+function sethome.go_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		-- just a check to prevent the server crashing
+		return false
+	end
+	if homepos[name] then
+		player:setpos(homepos[name])
+		minetest.chat_send_player(name, "Teleported to home!")
+		return true
```

Comment:
My objective was to let those functions be used as the chatcommand callback, I originally only moved the code from those callbacks into independant global functions.
For now the function returns both true/false and a message, acting by default as a chatcommand callback function. (requested [here](https://github.com/minetest/minetest_game/pull/747#discussion_r48088745))